{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Setup"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Connect to a local neo4j server"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Connection established.\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "from dotenv import load_dotenv\n",
    "from neo4j import GraphDatabase\n",
    "\n",
    "load_dotenv(\".env\")\n",
    "neo4j_auth = os.getenv(\"neo4j_pass\")\n",
    "\n",
    "URI = \"bolt://localhost:7687\"\n",
    "AUTH = (\"neo4j\", neo4j_auth)\n",
    "\n",
    "with GraphDatabase.driver(URI, auth=AUTH) as driver:\n",
    "    driver.verify_connectivity()\n",
    "    print(\"Connection established.\") #si no se establece la conexi√≥n crashea el programa\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Querying the database"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get all diseases from a community"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Record name='AIDS Dementia Complex'>\n",
      "<Record name='Alexia'>\n",
      "<Record name='Amnesia'>\n",
      "<Record name='Retrograde amnesia'>\n",
      "<Record name='Amnestic Disorder'>\n",
      "<Record name='Cognition Disorders'>\n",
      "<Record name='Delirium'>\n",
      "<Record name='Delusional disorder'>\n",
      "<Record name='Delirium, Dementia, Amnestic, Cognitive Disorders'>\n",
      "<Record name='Other and unspecified reactive psychosis'>\n",
      "<Record name='Psychoses, Alcoholic'>\n",
      "<Record name='Psychoses, Drug'>\n",
      "<Record name='Psychoses, Substance-Induced'>\n",
      "<Record name='Psychotic Disorders'>\n",
      "<Record name='Schizoaffective Disorder'>\n",
      "<Record name='Schizophreniform Disorders'>\n",
      "<Record name='Other specified senile psychotic conditions'>\n",
      "<Record name='Amnesia, Transient Global'>\n",
      "<Record name='Dementia'>\n",
      "<Record name='Lewy Body Disease'>\n",
      "<Record name='Other alcohol-induced mental disorders'>\n",
      "<Record name='Lewy Body Variant of Alzheimer Disease'>\n",
      "<Record name='Neurocognitive Disorders'>\n",
      "The query `MATCH (d:Disease {comunidad_infomap: $comunidad}) RETURN d.name AS name` returned 23 records in 38 ms.\n"
     ]
    }
   ],
   "source": [
    "with GraphDatabase.driver(URI, auth=AUTH) as driver:\n",
    "    # Get the name of all 42 year-olds\n",
    "    records, summary, keys = driver.execute_query(\n",
    "        \"MATCH (d:Disease {comunidad_infomap: $comunidad}) RETURN d.name AS name\",\n",
    "        comunidad=1,\n",
    "        database_=\"neo4j\",\n",
    "    )\n",
    "\n",
    "# Loop through results and do something with them\n",
    "for disease in records:\n",
    "    print(disease)\n",
    "\n",
    "# Summary information\n",
    "print(\"The query `{query}` returned {records_count} records in {time} ms.\".format(\n",
    "    query=summary.query, records_count=len(records),\n",
    "    time=summary.result_available_after,\n",
    "))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get all genes linked to a specific disease"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Record genes='PPARGC1A'>\n",
      "<Record genes='AGER'>\n",
      "<Record genes='EDN1'>\n",
      "<Record genes='AIF1'>\n",
      "<Record genes='ENO2'>\n",
      "<Record genes='GBA'>\n",
      "<Record genes='GFAP'>\n",
      "<Record genes='IGF1R'>\n",
      "<Record genes='IGF2'>\n",
      "<Record genes='IGF2R'>\n",
      "<Record genes='INS'>\n",
      "<Record genes='INSR'>\n",
      "<Record genes='MAG'>\n",
      "<Record genes='MAP2'>\n",
      "<Record genes='NGF'>\n",
      "<Record genes='NTRK1'>\n",
      "<Record genes='NTRK2'>\n",
      "<Record genes='SNCB'>\n",
      "<Record genes='SNCA'>\n",
      "<Record genes='SOD2'>\n",
      "<Record genes='TH'>\n",
      "<Record genes='BECN1'>\n",
      "The query `MATCH (d:Disease {name: $name}), (d)-[:ASSOCIATED_WITH]->(g:Gene)  RETURN g.name AS genes` returned 22 records in 2 ms.\n"
     ]
    }
   ],
   "source": [
    "with GraphDatabase.driver(URI, auth=AUTH) as driver:\n",
    "    # Get the name of all 42 year-olds\n",
    "    records, summary, keys = driver.execute_query(\n",
    "        \"MATCH (d:Disease {name: $name}), (d)-[:ASSOCIATED_WITH]->(g:Gene)  RETURN g.name AS genes\",\n",
    "        name=\"Lewy Body Disease\",\n",
    "        database_=\"neo4j\",\n",
    "    )\n",
    "\n",
    "# Loop through results and do something with them\n",
    "for disease in records:\n",
    "    print(disease)\n",
    "\n",
    "# Summary information\n",
    "print(\"The query `{query}` returned {records_count} records in {time} ms.\".format(\n",
    "    query=summary.query, records_count=len(records),\n",
    "    time=summary.result_available_after,\n",
    "))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get all genes linked with a disease community"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Record genes='ADRB2'>\n",
      "<Record genes='OPRM1'>\n",
      "<Record genes='DNAAF4'>\n",
      "<Record genes='SIGMAR1'>\n",
      "<Record genes='HRH3'>\n",
      "<Record genes='CSF2'>\n",
      "<Record genes='GNAI1'>\n",
      "<Record genes='GNAI3'>\n",
      "<Record genes='GRP'>\n",
      "<Record genes='APP'>\n",
      "<Record genes='IL1A'>\n",
      "<Record genes='IL2'>\n",
      "<Record genes='IL6'>\n",
      "<Record genes='NGF'>\n",
      "<Record genes='PDYN'>\n",
      "<Record genes='POMC'>\n",
      "<Record genes='AVP'>\n",
      "<Record genes='PREP'>\n",
      "<Record genes='TAC1'>\n",
      "<Record genes='TRH'>\n",
      "<Record genes='PREP'>\n",
      "<Record genes='ABCC4'>\n",
      "<Record genes='SLC51B'>\n",
      "<Record genes='COMT'>\n",
      "<Record genes='CRH'>\n",
      "<Record genes='DRD2'>\n",
      "<Record genes='DRD3'>\n",
      "<Record genes='AGT'>\n",
      "<Record genes='EIF2S1'>\n",
      "<Record genes='SLC51A'>\n",
      "<Record genes='EPO'>\n",
      "<Record genes='FGF14'>\n",
      "<Record genes='FMR1'>\n",
      "<Record genes='FOSB'>\n",
      "<Record genes='AFF4'>\n",
      "<Record genes='RAB40AL'>\n",
      "<Record genes='GSK3B'>\n",
      "<Record genes='APOE'>\n",
      "<Record genes='IGF2'>\n",
      "<Record genes='APP'>\n",
      "<Record genes='LAMB2'>\n",
      "<Record genes='SMAD4'>\n",
      "<Record genes='MAOA'>\n",
      "<Record genes='MAPT'>\n",
      "<Record genes='MET'>\n",
      "<Record genes='OTC'>\n",
      "<Record genes='OXTR'>\n",
      "<Record genes='PSEN1'>\n",
      "<Record genes='SLC4A10'>\n",
      "<Record genes='PTGS2'>\n",
      "<Record genes='BCHE'>\n",
      "<Record genes='SLC1A1'>\n",
      "<Record genes='SLC6A4'>\n",
      "<Record genes='SLC10A2'>\n",
      "<Record genes='TSC1'>\n",
      "<Record genes='VIP'>\n",
      "<Record genes='ZNF41'>\n",
      "<Record genes='NR0B2'>\n",
      "<Record genes='CDK5R1'>\n",
      "<Record genes='DRD2'>\n",
      "<Record genes='DRD3'>\n",
      "<Record genes='IGF1'>\n",
      "<Record genes='APOE'>\n",
      "<Record genes='IL1RN'>\n",
      "<Record genes='IL6'>\n",
      "<Record genes='CXCL8'>\n",
      "<Record genes='LIF'>\n",
      "<Record genes='SLC6A3'>\n",
      "<Record genes='TH'>\n",
      "<Record genes='DPYSL2'>\n",
      "<Record genes='INS'>\n",
      "<Record genes='ADORA2A'>\n",
      "<Record genes='ACE'>\n",
      "<Record genes='AKT1'>\n",
      "<Record genes='ESR1'>\n",
      "<Record genes='GCLM'>\n",
      "<Record genes='HPGDS'>\n",
      "<Record genes='GRM2'>\n",
      "<Record genes='GSTP1'>\n",
      "<Record genes='HTR1A'>\n",
      "<Record genes='HTR6'>\n",
      "<Record genes='NPY1R'>\n",
      "<Record genes='OPRM1'>\n",
      "<Record genes='PHB'>\n",
      "<Record genes='SLC6A3'>\n",
      "<Record genes='SNCA'>\n",
      "<Record genes='SOD2'>\n",
      "<Record genes='CALR'>\n",
      "<Record genes='CAMK2B'>\n",
      "<Record genes='DTNBP1'>\n",
      "<Record genes='RGS9'>\n",
      "<Record genes='PICK1'>\n",
      "<Record genes='ADORA2A'>\n",
      "<Record genes='ACE'>\n",
      "<Record genes='AKT1'>\n",
      "<Record genes='ESR1'>\n",
      "<Record genes='GRM2'>\n",
      "<Record genes='GSTP1'>\n",
      "<Record genes='HTR1A'>\n",
      "<Record genes='HTR6'>\n",
      "<Record genes='NPY1R'>\n",
      "<Record genes='OPRM1'>\n",
      "<Record genes='PHB'>\n",
      "<Record genes='SLC6A3'>\n",
      "<Record genes='SNCA'>\n",
      "<Record genes='SOD2'>\n",
      "<Record genes='CAMK2B'>\n",
      "<Record genes='RGS9'>\n",
      "<Record genes='PICK1'>\n",
      "<Record genes='RBM12'>\n",
      "<Record genes='CDH17'>\n",
      "<Record genes='OLIG2'>\n",
      "<Record genes='ADARB1'>\n",
      "<Record genes='CPLX2'>\n",
      "<Record genes='CPLX1'>\n",
      "<Record genes='PDE10A'>\n",
      "<Record genes='TPH2'>\n",
      "<Record genes='ADH1B'>\n",
      "<Record genes='CRHR1'>\n",
      "<Record genes='CSF2RB'>\n",
      "<Record genes='CYP2B6'>\n",
      "<Record genes='DLX1'>\n",
      "<Record genes='DRD5'>\n",
      "<Record genes='APOBEC3A'>\n",
      "<Record genes='C9orf72'>\n",
      "<Record genes='SP8'>\n",
      "<Record genes='FGFR2'>\n",
      "<Record genes='FKBP5'>\n",
      "<Record genes='ARHGAP26'>\n",
      "<Record genes='WWC1'>\n",
      "<Record genes='FTL'>\n",
      "<Record genes='TENM4'>\n",
      "<Record genes='GCH1'>\n",
      "<Record genes='HPGDS'>\n",
      "<Record genes='APOBEC3C'>\n",
      "<Record genes='GLUL'>\n",
      "<Record genes='GNA12'>\n",
      "<Record genes='ANK3'>\n",
      "<Record genes='GRIK3'>\n",
      "<Record genes='GRIK4'>\n",
      "<Record genes='GRIN1'>\n",
      "<Record genes='NR3C1'>\n",
      "<Record genes='GSTA1'>\n",
      "<Record genes='GSTP1'>\n",
      "<Record genes='SLC39A3'>\n",
      "<Record genes='HDAC2'>\n",
      "<Record genes='HLA-B'>\n",
      "<Record genes='HLA-E'>\n",
      "<Record genes='HTR1B'>\n",
      "<Record genes='HTR2A'>\n",
      "<Record genes='HTR5A'>\n",
      "<Record genes='KCNH2'>\n",
      "<Record genes='LEP'>\n",
      "<Record genes='MAOA'>\n",
      "<Record genes='ME2'>\n",
      "<Record genes='MEF2A'>\n",
      "<Record genes='LMOD2'>\n",
      "<Record genes='GADD45B'>\n",
      "<Record genes='SEPTIN2'>\n",
      "<Record genes='NQO2'>\n",
      "<Record genes='NPAS1'>\n",
      "<Record genes='NPY1R'>\n",
      "<Record genes='NTRK1'>\n",
      "<Record genes='OPRM1'>\n",
      "<Record genes='PAH'>\n",
      "<Record genes='PAWR'>\n",
      "<Record genes='SHANK1'>\n",
      "<Record genes='PCNT'>\n",
      "<Record genes='ABCB1'>\n",
      "<Record genes='DGCR8'>\n",
      "<Record genes='GNB1L'>\n",
      "<Record genes='NDE1'>\n",
      "<Record genes='SOBP'>\n",
      "<Record genes='AVP'>\n",
      "<Record genes='PBRM1'>\n",
      "<Record genes='PPP3CC'>\n",
      "<Record genes='PRKCA'>\n",
      "<Record genes='ST6GALNAC1'>\n",
      "<Record genes='PAK5'>\n",
      "<Record genes='PTPRC'>\n",
      "<Record genes='PVALB'>\n",
      "<Record genes='BCL9'>\n",
      "<Record genes='BDNF'>\n",
      "<Record genes='SAT1'>\n",
      "<Record genes='NPAS3'>\n",
      "<Record genes='ZSCAN31'>\n",
      "<Record genes='ST3GAL1'>\n",
      "<Record genes='SLC6A9'>\n",
      "<Record genes='SLC12A2'>\n",
      "<Record genes='SNAP25'>\n",
      "<Record genes='SP1'>\n",
      "<Record genes='TBX1'>\n",
      "<Record genes='TCF4'>\n",
      "<Record genes='TDO2'>\n",
      "<Record genes='TGFB1'>\n",
      "<Record genes='TRPM2'>\n",
      "<Record genes='TYR'>\n",
      "<Record genes='CACNA1C'>\n",
      "<Record genes='TET1'>\n",
      "<Record genes='KCNH6'>\n",
      "<Record genes='CALR'>\n",
      "<Record genes='NDEL1'>\n",
      "<Record genes='PLA2G6'>\n",
      "<Record genes='CCKAR'>\n",
      "<Record genes='SELENBP1'>\n",
      "<Record genes='GPR52'>\n",
      "<Record genes='PICK1'>\n",
      "<Record genes='CD34'>\n",
      "<Record genes='FEZ1'>\n",
      "<Record genes='MED12'>\n",
      "<Record genes='ERVW-4'>\n",
      "<Record genes='RBM12'>\n",
      "<Record genes='CPLX2'>\n",
      "<Record genes='CPLX1'>\n",
      "<Record genes='DRD5'>\n",
      "<Record genes='C9orf72'>\n",
      "<Record genes='SULT4A1'>\n",
      "<Record genes='GRIA1'>\n",
      "<Record genes='GRIN1'>\n",
      "<Record genes='NR3C1'>\n",
      "<Record genes='HTR1B'>\n",
      "<Record genes='HTR2A'>\n",
      "<Record genes='NEUROG1'>\n",
      "<Record genes='NPY'>\n",
      "<Record genes='NPAS2'>\n",
      "<Record genes='NTRK1'>\n",
      "<Record genes='MED15'>\n",
      "<Record genes='PDGFRB'>\n",
      "<Record genes='PLA2G4A'>\n",
      "<Record genes='AVP'>\n",
      "<Record genes='PRKCA'>\n",
      "<Record genes='PVALB'>\n",
      "<Record genes='BDNF'>\n",
      "<Record genes='SAT1'>\n",
      "<Record genes='SNAP25'>\n",
      "<Record genes='SST'>\n",
      "<Record genes='TCF7L2'>\n",
      "<Record genes='CALR'>\n",
      "<Record genes='SELENBP1'>\n",
      "<Record genes='RBM12'>\n",
      "<Record genes='CPLX2'>\n",
      "<Record genes='CPLX1'>\n",
      "<Record genes='DRD5'>\n",
      "<Record genes='GRIN1'>\n",
      "<Record genes='NR3C1'>\n",
      "<Record genes='HTR1B'>\n",
      "<Record genes='HTR2A'>\n",
      "<Record genes='NTRK1'>\n",
      "<Record genes='AVP'>\n",
      "<Record genes='PVALB'>\n",
      "<Record genes='BDNF'>\n",
      "<Record genes='SNAP25'>\n",
      "<Record genes='SELENBP1'>\n",
      "<Record genes='COL4A1'>\n",
      "<Record genes='CP'>\n",
      "<Record genes='CSF1R'>\n",
      "<Record genes='CST3'>\n",
      "<Record genes='DNMT1'>\n",
      "<Record genes='SLC9A8'>\n",
      "<Record genes='FUS'>\n",
      "<Record genes='GRN'>\n",
      "<Record genes='APP'>\n",
      "<Record genes='MAPT'>\n",
      "<Record genes='NOTCH3'>\n",
      "<Record genes='TREM2'>\n",
      "<Record genes='PRNP'>\n",
      "<Record genes='HTRA1'>\n",
      "<Record genes='PSEN2'>\n",
      "<Record genes='SLC6A3'>\n",
      "<Record genes='TYROBP'>\n",
      "<Record genes='PPARGC1A'>\n",
      "<Record genes='AGER'>\n",
      "<Record genes='EDN1'>\n",
      "<Record genes='AIF1'>\n",
      "<Record genes='ENO2'>\n",
      "<Record genes='GBA'>\n",
      "<Record genes='GFAP'>\n",
      "<Record genes='IGF1R'>\n",
      "<Record genes='IGF2'>\n",
      "<Record genes='IGF2R'>\n",
      "<Record genes='INS'>\n",
      "<Record genes='INSR'>\n",
      "<Record genes='MAG'>\n",
      "<Record genes='MAP2'>\n",
      "<Record genes='NGF'>\n",
      "<Record genes='NTRK1'>\n",
      "<Record genes='NTRK2'>\n",
      "<Record genes='SNCB'>\n",
      "<Record genes='SNCA'>\n",
      "<Record genes='SOD2'>\n",
      "<Record genes='TH'>\n",
      "<Record genes='BECN1'>\n",
      "<Record genes='INS'>\n",
      "The query `MATCH (d:Disease {comunidad_infomap: $comunidad}), (d)-[:ASSOCIATED_WITH]->(g:Gene)  RETURN g.name AS genes` returned 293 records in 59 ms.\n"
     ]
    }
   ],
   "source": [
    "with GraphDatabase.driver(URI, auth=AUTH) as driver:\n",
    "    # Get the name of all 42 year-olds\n",
    "    records, summary, keys = driver.execute_query(\n",
    "        \"MATCH (d:Disease {comunidad_infomap: $comunidad}), (d)-[:ASSOCIATED_WITH]->(g:Gene)  RETURN g.name AS genes\",\n",
    "        comunidad=1,\n",
    "        database_=\"neo4j\",\n",
    "    )\n",
    "\n",
    "# Loop through results and do something with them\n",
    "for disease in records:\n",
    "    print(disease)\n",
    "\n",
    "# Summary information\n",
    "print(\"The query `{query}` returned {records_count} records in {time} ms.\".format(\n",
    "    query=summary.query, records_count=len(records),\n",
    "    time=summary.result_available_after,\n",
    "))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get all protein complexes formed by the proteins (codified by the genes) associated with a specific disease"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Record gene='BECN1' complex='Vps34 Complex II'>\n",
      "<Record gene='BECN1' complex='Vps34 Complex I'>\n",
      "The query `MATCH (d:Disease {name: $name}), (d)-[:ASSOCIATED_WITH]->(g:Gene), (g)-[:PARTICIPATES_IN]->(c:Complex)  RETURN g.name AS gene, c.name AS complex` returned 2 records in 89 ms.\n"
     ]
    }
   ],
   "source": [
    "with GraphDatabase.driver(URI, auth=AUTH) as driver:\n",
    "    # Get the name of all 42 year-olds\n",
    "    records, summary, keys = driver.execute_query(\n",
    "        \"MATCH (d:Disease {name: $name}), (d)-[:ASSOCIATED_WITH]->(g:Gene), (g)-[:PARTICIPATES_IN]->(c:Complex)  RETURN g.name AS gene, c.name AS complex\",\n",
    "        name=\"Lewy Body Disease\",\n",
    "        database_=\"neo4j\",\n",
    "    )\n",
    "\n",
    "# Loop through results and do something with them\n",
    "for disease in records:\n",
    "    print(disease)\n",
    "\n",
    "# Summary information\n",
    "print(\"The query `{query}` returned {records_count} records in {time} ms.\".format(\n",
    "    query=summary.query, records_count=len(records),\n",
    "    time=summary.result_available_after,\n",
    "))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get all biological pathways in which the proteins (codified by the genes) associated with a disease participate in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Record gene='PPARGC1A' pathway='PPARA activates gene expression'>\n",
      "<Record gene='PPARGC1A' pathway='Transcriptional activation of mitochondrial biogenesis'>\n",
      "<Record gene='PPARGC1A' pathway='Activation of PPARGC1A (PGC-1alpha) by phosphorylation'>\n",
      "<Record gene='PPARGC1A' pathway='Transcriptional regulation of white adipocyte differentiation'>\n",
      "<Record gene='PPARGC1A' pathway='SUMOylation of transcription cofactors'>\n",
      "<Record gene='PPARGC1A' pathway='Circadian Clock'>\n",
      "<Record gene='PPARGC1A' pathway='Regulation of RUNX2 expression and activity'>\n",
      "<Record gene='PPARGC1A' pathway='FOXO-mediated transcription of oxidative stress, metabolic and neuronal genes'>\n",
      "<Record gene='PPARGC1A' pathway='Heme signaling'>\n",
      "<Record gene='AGER' pathway='TAK1 activates NFkB by phosphorylation and activation of IKKs complex'>\n",
      "<Record gene='AGER' pathway='Advanced glycosylation endproduct receptor signaling'>\n",
      "<Record gene='AGER' pathway='TRAF6 mediated NF-kB activation'>\n",
      "<Record gene='EDN1' pathway='Peptide ligand-binding receptors'>\n",
      "<Record gene='EDN1' pathway='G alpha (q) signalling events'>\n",
      "<Record gene='ENO2' pathway='Glycolysis'>\n",
      "<Record gene='ENO2' pathway='Gluconeogenesis'>\n",
      "<Record gene='GBA' pathway='Glycosphingolipid metabolism'>\n",
      "<Record gene='GBA' pathway='Association of TriC/CCT with target proteins during biosynthesis'>\n",
      "<Record gene='GFAP' pathway='Nuclear signaling by ERBB4'>\n",
      "<Record gene='GFAP' pathway='Chaperone Mediated Autophagy'>\n",
      "<Record gene='IGF1R' pathway='Signaling by Type 1 Insulin-like Growth Factor 1 Receptor (IGF1R)'>\n",
      "<Record gene='IGF1R' pathway='IRS-related events triggered by IGF1R'>\n",
      "<Record gene='IGF1R' pathway='SHC-related events triggered by IGF1R'>\n",
      "<Record gene='IGF1R' pathway='Extra-nuclear estrogen signaling'>\n",
      "<Record gene='IGF2' pathway='Platelet degranulation '>\n",
      "<Record gene='IGF2' pathway='Signaling by Type 1 Insulin-like Growth Factor 1 Receptor (IGF1R)'>\n",
      "<Record gene='IGF2' pathway='IRS-related events triggered by IGF1R'>\n",
      "<Record gene='IGF2' pathway='SHC-related events triggered by IGF1R'>\n",
      "<Record gene='IGF2' pathway='Regulation of Insulin-like Growth Factor (IGF) transport and uptake by Insulin-like Growth Factor Binding Proteins (IGFBPs)'>\n",
      "<Record gene='IGF2R' pathway='Golgi Associated Vesicle Biogenesis'>\n",
      "<Record gene='IGF2R' pathway='Neutrophil degranulation'>\n",
      "<Record gene='IGF2R' pathway='Retrograde transport at the Trans-Golgi-Network'>\n",
      "<Record gene='IGF2R' pathway='Cargo recognition for clathrin-mediated endocytosis'>\n",
      "<Record gene='IGF2R' pathway='Clathrin-mediated endocytosis'>\n",
      "<Record gene='INS' pathway='Regulation of gene expression in beta cells'>\n",
      "<Record gene='INS' pathway='Insulin processing'>\n",
      "<Record gene='INS' pathway='Synthesis, secretion, and deacylation of Ghrelin'>\n",
      "<Record gene='INS' pathway='Regulation of insulin secretion'>\n",
      "<Record gene='INS' pathway='COPI-mediated anterograde transport'>\n",
      "<Record gene='INS' pathway='PI5P, PP2A and IER3 Regulate PI3K/AKT Signaling'>\n",
      "<Record gene='INS' pathway='IRS activation'>\n",
      "<Record gene='INS' pathway='Signal attenuation'>\n",
      "<Record gene='INS' pathway='Insulin receptor signalling cascade'>\n",
      "<Record gene='INS' pathway='Signaling by Insulin receptor'>\n",
      "<Record gene='INS' pathway='Insulin receptor recycling'>\n",
      "<Record gene='INS' pathway='FOXO-mediated transcription of oxidative stress, metabolic and neuronal genes'>\n",
      "<Record gene='INS' pathway='Amyloid fiber formation'>\n",
      "<Record gene='INSR' pathway='PI5P, PP2A and IER3 Regulate PI3K/AKT Signaling'>\n",
      "<Record gene='INSR' pathway='IRS activation'>\n",
      "<Record gene='INSR' pathway='Signal attenuation'>\n",
      "<Record gene='INSR' pathway='Insulin receptor signalling cascade'>\n",
      "<Record gene='INSR' pathway='Signaling by Insulin receptor'>\n",
      "<Record gene='INSR' pathway='Insulin receptor recycling'>\n",
      "<Record gene='MAG' pathway='Axonal growth inhibition (RHOA activation)'>\n",
      "<Record gene='MAG' pathway='Basigin interactions'>\n",
      "<Record gene='MAG' pathway='EGR2 and SOX10-mediated initiation of Schwann cell myelination'>\n",
      "<Record gene='NGF' pathway='PLC-gamma1 signalling'>\n",
      "<Record gene='NGF' pathway='Signalling to RAS'>\n",
      "<Record gene='NGF' pathway='NGF processing'>\n",
      "<Record gene='NGF' pathway='Frs2-mediated activation'>\n",
      "<Record gene='NGF' pathway='ARMS-mediated activation'>\n",
      "<Record gene='NGF' pathway='Retrograde neurotrophin signalling'>\n",
      "<Record gene='NGF' pathway='TRKA activation by NGF'>\n",
      "<Record gene='NGF' pathway='Signalling to p38 via RIT and RIN'>\n",
      "<Record gene='NGF' pathway='NRAGE signals death through JNK'>\n",
      "<Record gene='NGF' pathway='p75NTR negatively regulates cell cycle via SC1'>\n",
      "<Record gene='NGF' pathway='Ceramide signalling'>\n",
      "<Record gene='NGF' pathway='PI3K/AKT activation'>\n",
      "<Record gene='NGF' pathway='Signalling to STAT3'>\n",
      "<Record gene='NGF' pathway='NFG and proNGF binds to p75NTR'>\n",
      "<Record gene='NGF' pathway='NADE modulates death signalling'>\n",
      "<Record gene='NGF' pathway='NRIF signals cell death from the nucleus'>\n",
      "<Record gene='NGF' pathway='p75NTR recruits signalling complexes'>\n",
      "<Record gene='NGF' pathway='NF-kB is activated and signals survival'>\n",
      "<Record gene='NGF' pathway='Axonal growth stimulation'>\n",
      "<Record gene='NTRK1' pathway='PLC-gamma1 signalling'>\n",
      "<Record gene='NTRK1' pathway='Signalling to RAS'>\n",
      "<Record gene='NTRK1' pathway='Frs2-mediated activation'>\n",
      "<Record gene='NTRK1' pathway='ARMS-mediated activation'>\n",
      "<Record gene='NTRK1' pathway='Retrograde neurotrophin signalling'>\n",
      "<Record gene='NTRK1' pathway='NGF-independant TRKA activation'>\n",
      "<Record gene='NTRK1' pathway='TRKA activation by NGF'>\n",
      "<Record gene='NTRK1' pathway='Signalling to p38 via RIT and RIN'>\n",
      "<Record gene='NTRK1' pathway='PI3K/AKT activation'>\n",
      "<Record gene='NTRK1' pathway='Signalling to STAT3'>\n",
      "<Record gene='NTRK2' pathway='NGF-independant TRKA activation'>\n",
      "<Record gene='NTRK2' pathway='BDNF activates NTRK2 (TRKB) signaling'>\n",
      "<Record gene='NTRK2' pathway='NTF3 activates NTRK2 (TRKB) signaling'>\n",
      "<Record gene='NTRK2' pathway='NTF4 activates NTRK2 (TRKB) signaling'>\n",
      "<Record gene='NTRK2' pathway='Activated NTRK2 signals through RAS'>\n",
      "<Record gene='NTRK2' pathway='Activated NTRK2 signals through PLCG1'>\n",
      "<Record gene='NTRK2' pathway='Activated NTRK2 signals through PI3K'>\n",
      "<Record gene='NTRK2' pathway='Activated NTRK2 signals through FRS2 and FRS3'>\n",
      "<Record gene='NTRK2' pathway='Activated NTRK2 signals through FYN'>\n",
      "<Record gene='NTRK2' pathway='NTRK2 activates RAC1'>\n",
      "<Record gene='NTRK2' pathway='Activated NTRK2 signals through CDK5'>\n",
      "<Record gene='SNCB' pathway='MTF1 activates gene expression'>\n",
      "<Record gene='SNCA' pathway='Amyloid fiber formation'>\n",
      "<Record gene='SOD2' pathway='Transcriptional activation of mitochondrial biogenesis'>\n",
      "<Record gene='SOD2' pathway='Detoxification of Reactive Oxygen Species'>\n",
      "<Record gene='SOD2' pathway=\"Deregulated CDK5 triggers multiple neurodegenerative pathways in Alzheimer's disease models\">\n",
      "<Record gene='SOD2' pathway='Gene and protein expression by JAK-STAT signaling after Interleukin-12 stimulation'>\n",
      "<Record gene='SOD2' pathway='FOXO-mediated transcription of oxidative stress, metabolic and neuronal genes'>\n",
      "<Record gene='TH' pathway='Catecholamine biosynthesis'>\n",
      "<Record gene='BECN1' pathway='Macroautophagy'>\n",
      "<Record gene='BECN1' pathway='Ub-specific processing proteases'>\n",
      "<Record gene='BECN1' pathway='Translation of Replicase and Assembly of the Replication Transcription Complex'>\n",
      "<Record gene='BECN1' pathway='Translation of Replicase and Assembly of the Replication Transcription Complex'>\n",
      "The query `MATCH (d:Disease {name: $name}), (d)-[:ASSOCIATED_WITH]->(g:Gene), (g)-[:PARTICIPATES_IN]->(p:Pathway)  RETURN g.name AS gene, p.name AS pathway` returned 108 records in 65 ms.\n"
     ]
    }
   ],
   "source": [
    "with GraphDatabase.driver(URI, auth=AUTH) as driver:\n",
    "    # Get the name of all 42 year-olds\n",
    "    records, summary, keys = driver.execute_query(\n",
    "        \"MATCH (d:Disease {name: $name}), (d)-[:ASSOCIATED_WITH]->(g:Gene), (g)-[:PARTICIPATES_IN]->(p:Pathway)  RETURN g.name AS gene, p.name AS pathway\",\n",
    "        name=\"Lewy Body Disease\",\n",
    "        database_=\"neo4j\",\n",
    "    )\n",
    "\n",
    "# Loop through results and do something with them\n",
    "for disease in records:\n",
    "    print(disease)\n",
    "\n",
    "# Summary information\n",
    "print(\"The query `{query}` returned {records_count} records in {time} ms.\".format(\n",
    "    query=summary.query, records_count=len(records),\n",
    "    time=summary.result_available_after,\n",
    "))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
